mungegithub/mungers/milestone-maintainer_test.go

```diff
@@ -0,0 +1,233 @@
+	}
+
+	server.Close()
+}
+
+func milestoneTestComment(label string, createdAt time.Time) *c.Comment {
+	n := &c.Notification{
+		Name:      milestoneNotifierName,
+		Arguments: label,
+		Context:   "foo",
+	}
+	return &c.Comment{
+		Body:      stringPtr(n.String()),
+		CreatedAt: &createdAt,
+	}
+}
+
+func milestoneLabelEvents(label string, createdAt time.Time) []*githubapi.IssueEvent {
+	return []*githubapi.IssueEvent{
+		{
+			Event: stringPtr("labeled"),
+			Label: &githubapi.Label{
+				Name: &label,
+			},
+			CreatedAt: &createdAt,
+			Actor: &githubapi.User{
+				Login: stringPtr(milestoneTestBotName),
+			},
+		},
+	}
+}
+
+// TestNotificationState validates how the notificationState() method
+// responds to the possible label and notification states of issues in
+// a milestone.
+func TestNotificationState(t *testing.T) {
+	createdNow := time.Now()
+	createdYesterday := createdNow.Add(-(milestoneWarningPeriod + time.Hour))
+	createdLongAgo := createdNow.Add(-(milestoneRemovalPeriod + time.Hour))
+	tests := map[string]struct {
+		labels                 []string
+		comment                *c.Comment
+		events                 []*githubapi.IssueEvent
+		expectedLabel          string
+		expectedDeleteExisting bool
+		expectedDescription    string
+	}{
+		"Label and comment with summary for complete labels": {
+			labels:              []string{"kind/bug", "priority/important-soon", "sig/foo"},
+			expectedLabel:       milestoneLabelsCompleteLabel,
+			expectedDescription: milestoneLabelsComplete,
+		},
+		"Do nothing for up-to-date summary": {
+			labels:        []string{"kind/bug", "priority/important-soon", "sig/foo"},
+			comment:       milestoneTestComment(milestoneLabelsComplete, createdNow),
+			expectedLabel: milestoneLabelsCompleteLabel,
+		},
+		"Label and warn for the first time": {
+			expectedLabel:       milestoneLabelsIncompleteLabel,
+			expectedDescription: milestoneLabelsIncomplete,
+		},
+		"Do nothing for up-to-date warning": {
+			comment:       milestoneTestComment(milestoneLabelsIncomplete, createdNow),
+			expectedLabel: milestoneLabelsIncompleteLabel,
+		},
+		"Warn again after warning interval has elapsed": {
+			comment:                milestoneTestComment(milestoneLabelsIncomplete, createdYesterday),
+			expectedLabel:          milestoneLabelsIncompleteLabel,
+			expectedDeleteExisting: true,
+			expectedDescription:    milestoneLabelsIncomplete,
+		},
+		"Non-blocker is removed from milestone after grace period": {
+			labels:                 []string{milestoneLabelsIncompleteLabel},
+			comment:                milestoneTestComment(milestoneLabelsIncomplete, createdYesterday),
+			events:                 milestoneLabelEvents(milestoneLabelsIncompleteLabel, createdLongAgo),
+			expectedLabel:          milestoneRemovedLabel,
+			expectedDeleteExisting: true,
+			expectedDescription:    milestoneRemoved,
+		},
+		"Blocker is not removed from milestone after grace period": {
```

Comment:
Was that your intent, to allow user comment to influence milestone inclusion during the development cycle?  My reading of the process doc was that user feedback would be required to remain in the milestone only during slush and freeze (still to come).